### ISSUE TITLE
Engine examples builds twice on engine change when using ES6 build

### ISSUE THREAD

**Issue by mvaligursky** [playcanvas/engine#5384]
Command to rebuild engine on change:
```
target=debug_es6 npm run watch
```

Command to rebuild examples using module engine build:
```
ENGINE_PATH=../build/playcanvas.dbg.mjs/index.js npm run develop
```

When a file in the engine folder changes, engine rebuilds, and it seems the examples rebuilds twice:
```
[0] [0] bundles src/iframe/index.mjs → dist/iframe...
[0] [0]  Finished at: 09:34:35
[0] [0] created dist/iframe in 5.4s
[0] [0] bundles src/iframe/index.mjs → dist/iframe...
[0] [0]  Finished at: 09:34:40
[0] [0] created dist/iframe in 5.2s
```

Is my understanding correct that it does the same build twice? If so, can this be avoided.

**Comment by slimbuck** [comment]
The same happens for model-viewer and our other engine apps. Would also love to see it solved too (I've not had a chance to look into the cause myself).

**Comment by mvaligursky** [comment]
Not that this does not seem to happen with non-modules build of the engine. Is it the problem with the module build creating / updating multiple files in the build folder, each of those triggering the build of examples?

**Comment by mvaligursky** [comment]
This has been fixed in #5555
